version: "3"
name: linguacafe

networks:
    linguacafe:
        driver: bridge

services:
    webserver:
        container_name: linguacafe-webserver
        depends_on:
            mysql:
                condition: service_healthy
        volumes:
            - /your/linguacafe/dict/folder:/var/www/html/storage/app/dictionaries
            - /your/linguacafe/logs/folder:/var/www/html/storage/logs
        ports:
            - "9191:80"
            - "3000:3000"
        networks:
            - linguacafe
        extra_hosts:
            - "host.docker.internal:host-gateway"
    mysql:
        image: mysql:5.7.22
        container_name: linguacafe-database
        restart: unless-stopped
        tty: true
        ports:
            - "3308:3306"
        volumes:
            - /your/linguacafe/database/folder:/var/lib/mysql
        environment:
            MYSQL_DATABASE: linguacafe
            MYSQL_USER: linguacafe
            MYSQL_PASSWORD: linguacafe
            MYSQL_ROOT_PASSWORD: linguacafe
            SERVICE_NAME: linguacafe
        healthcheck:
            test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
            interval: 15s
            timeout: 15s
            retries: 8
        command: mysqld --general-log=1 --general-log-file=/var/lib/mysql/general-log.log
        networks:
            - linguacafe
    python:
        container_name: linguacafe-python-service
        command: /app/manage.py runserver 0.0.0.0:8678
        restart: unless-stopped
        tty: true
        volumes:
            - /your/linguacafe/dict/folder:/var/www/html/storage/app/dictionaries
        networks:
            - linguacafe
